mula/scheduler/schedulers/normalizer.py

```diff
@@ -40,172 +40,165 @@ def __init__(
+        listener = connectors.listeners.RawData(
+            dsn=self.ctx.config.host_raw_data,
+            queue=f"{self.organisation.id}__raw_file_received",
+            func=self.push_tasks_for_received_raw_data,
+        )
+        listener.listen()
 
-            latest_raw_data = None
-            try:
-                latest_raw_data = self.ctx.services.raw_data.get_latest_raw_data(
-                    queue=f"{self.organisation.id}__raw_file_received",
-                )
-            except (
-                pika.exceptions.ConnectionClosed,
-                pika.exceptions.ChannelClosed,
-                pika.exceptions.ChannelClosedByBroker,
-                pika.exceptions.AMQPConnectionError,
-            ) as e:
-                self.logger.debug(
-                    "Could not connect to rabbitmq queue: %s [organisation.id=%s, scheduler_id=%s]",
-                    f"{self.organisation.id}__raw_file_received",
-                    self.organisation.id,
-                    self.scheduler_id,
-                )
-                if self.stop_event.is_set():
-                    raise e
+    def push_tasks_for_received_raw_data(self, latest_raw_data: RawData) -> None:
+        if latest_raw_data is None:
+            self.logger.debug(
+                "No new raw data on message queue [organisation.id=%s, scheduler_id=%s]",
+                self.organisation.id,
+                self.scheduler_id,
+            )
+            return
 
-            # Stop the loop when we've processed everything from the
-            # messaging queue, so we can continue to the next step.
-            if latest_raw_data is None:
-                self.logger.debug(
-                    "No new raw data on message queue [organisation.id=%s, scheduler_id=%s]",
+        self.logger.debug(
+            "Received new raw data from message queue [raw_data.id=%s, organisation.id=%s, scheduler_id=%s]",
+            latest_raw_data.raw_data.id,
+            self.organisation.id,
+            self.scheduler_id,
+        )
+
+        # Check if the raw data doesn't contain an error mime-type,
+        # we don't need to create normalizers when the raw data returned
+        # an error.
+        for mime_type in latest_raw_data.raw_data.mime_types:
+            if mime_type.get("value", "").startswith("error/"):
+                self.logger.warning(
+                    "Skipping raw data with error mime type [raw_data.id=%s ,organisation.id=%s, scheduler_id=%s]",
+                    latest_raw_data.raw_data.id,
                     self.organisation.id,
                     self.scheduler_id,
                 )
                 return
 
+        # TODO: duplicates possible here?
+        normalizers = []
+        for mime_type in latest_raw_data.raw_data.mime_types:
+            normalizers_by_mime_type = self.get_normalizers_for_mime_type(mime_type.get("value"))
+            if not normalizers_by_mime_type:
+                continue
```

Comment:
Assuming that the function above can return an empty list, we don't necessarily need to check this emptiness when extending the other list

```suggestion
```